content-security-policy/support/testharness-helper.sub.js

```diff
@@ -41,6 +41,13 @@ function generateUrlWithPolicies(host, policy) {
     return url + "?policy=" + encodeURIComponent(policy);
 }
 
+function generateUrlWithAllowCSPFrom(host, allowCspFrom) {
+  var url = generateURL(host, PolicyHeader.ALLOW_CSP_FROM);
+  if (allowCspFrom == null)
+    return url + "?";
+  return url + "?allow_csp_from=" + encodeURIComponent(allowCspFrom);
```

Comment:
We can simplify this:

```
 var url = generateURL(host, PolicyHeader.ALLOW_CSP_FROM);
 if (allowCspFrom != null)
   url.searchParams.append("allow_csp_from", allowCspFrom);
 return url;
```

`URLSearchParams` will take care of the encoding for you; sorry I didn't think of that yesterday.

Likewise, you can update the other GET string operations to use that interface on the URL object. That should make the logic easier to follow.